atoma-service/src/handlers/mod.rs

```diff
@@ -602,7 +622,7 @@ mod vllm_metrics {
             model = model,
             best_url = best_url
         );
-        if min_request_queue_time_seconds >= MAX_ALLOWED_REQUEST_QUEUE_TIME_SECONDS {
+        if min_request_queue_time_seconds > MAX_ALLOWED_REQUEST_QUEUE_TIME_SECONDS {
```

Comment:
I don't think we should do this, since we are always updating for inferior values, so if all the vllm instances have higher waiting times for the queue, this will never be triggered

Reply:
The `min_request_queue_time_seconds` would be set to the current vllm value along as it's not `nan` , what we found is it wasn't getting updated and thus remained at 2 and triggered this error, whilst that may be due the prom query error due to overload, I think it's still safer to wait till it passes the threshold.

Reply:
The problem was (and still is) that if the prometheus query fails the metrics will be just Vec of errors. So that the `min_request_queue_time_seconds` is never updated.
We should not update the metrics_cache when the query itself fails.

Reply:
Good point @Cifko updated in 9586e44

Reply:
I don't think this will make it, as this logic will never be triggered.  We will need to change the logic of the current method.

Reply:
And how about the `min_request_queue_time_seconds` it will never be above the `MAX_ALLOWED_REQUEST_QUEUE_TIME_SECONDS` because we update only to smaller values than the initial.
Simply said, this will never trigger. So maybe set it to None and if we found a value smaller or the `min_request_queue_time_seconds.unwrap()` is greater then update it. That way we will get the actual waiting time in the logs.

Reply:
@Cifko I don't it should be set to `None` but rather `Infinity` which guaranteed to be greater than any real number, so the first valid request queue time will always be less than it (given it's a number and not `NaN` 

6090fa5